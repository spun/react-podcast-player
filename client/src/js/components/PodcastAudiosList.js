import React from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";

// Componente que muestra una lista de audios y controla sus acciones
const PodcastAudiosList = ({
	match,
	list,
	loadMoreAudios,
	playAudio,
	pauseAudio,
	markAudioAsPlayed,
	markAudioAsPending,
	currentAudioIdPlaying,
	isCurrentAudioPlaying
}) => {
	// Generar la fecha de publicación del audio corta
	function getMinimalDate(dateString) {
		var date = new Date(dateString);
		var monthString = date.toLocaleString("en-us", {
			month: "short"
		});
		var dayString = date.toLocaleString("en-us", {
			day: "numeric"
		});

		return (
			<div className="text-center" title={date.toString()}>
				<span className="audio-date small ">
					{monthString}
				</span>
				<br />
				<span className="audio-date small ">
					{dayString}
				</span>
			</div>
		);
	}

	return (
		<div className="card">
			{/* Titulo de la lista */}
			<div className="header">
				<h4 className="title">Podcast audios</h4>
			</div>

			{/* Lista de audios*/}
			<div className="content">
				<ul className="list-unstyled team-members">
					{/* Comprobar si la lista está vacía*/}
					{list.length == 0
						? <p className="text-info text-center">
								This podcast has no audios
							</p>
						: list.map((audio, i) =>
								<li key={i}>
									<div className="row">
										<div className="col-xs-1">
											{getMinimalDate(audio.audio_pubdate)}
										</div>
										<div className="col-xs-9">
											<NavLink
												to={`${match.url}/audios/${audio.audio_id}`}
												activeStyle={{
													fontWeight: "bold",
													color: "green"
												}}>
												{audio.audio_title}
											</NavLink>
											<br />

											{/*menú de marcar como escuchado/pendiente*/}
											<span
												className={
													audio.listen_status ==
													"pending"
														? "text-success audio-status audio-status-pending"
														: "text-muted audio-status"
												}
												data-toggle="dropdown">
												<small>
													{audio.listen_status ==
													"pending"
														? "Pending "
														: "Played "}

													<span className="caret" />
												</small>
											</span>
											<ul className="dropdown-menu">
												{audio.listen_status ==
													"pending" &&
													<li
														onClick={e => {
															markAudioAsPlayed(
																audio
															);
														}}>
														<a href="#">
															Mark as played
														</a>
													</li>}
												{audio.listen_status !=
													"pending" &&
													<li
														onClick={e =>
															markAudioAsPending(
																audio
															)}>
														<a href="#">
															Mark as unplayed
														</a>
													</li>}
												<li>
													<a
														href={audio.audio_url}
														download>
														Download file
													</a>
												</li>
											</ul>
										</div>
										{/*Botón de reproducir*/}
										<div className="col-xs-2 text-right">
											{currentAudioIdPlaying !=
												audio.audio_id ||
											isCurrentAudioPlaying == false
												? <button
														className="btn btn-sm btn-success btn-icon"
														onClick={e =>
															playAudio(audio)}>
														<i className="ti-control-play" />
													</button>
												: <button
														className="btn btn-sm btn-success btn-icon"
														onClick={e => pauseAudio()}>
														<i className="ti-control-pause" />
													</button>}
										</div>
									</div>
								</li>
							)}
				</ul>
				{/*Botón de cargar más audios*/}
				<div className=" footer text-center">
					<button
						type="submit"
						className="btn btn-info btn-fill btn-wd"
						onClick={e => loadMoreAudios()}>
						Load more audios
					</button>
				</div>
			</div>
		</div>
	);
};
export default PodcastAudiosList;
