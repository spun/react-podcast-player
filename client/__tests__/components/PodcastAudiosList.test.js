import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
// Component under test
import PodcastAudiosList from "../../src/js/components/PodcastAudiosList";

test("PodcastAudiosList creation", () => {
	const audios = [
		{ audio_id: 1, audio_title: "audio 1" },
		{ audio_id: 2, audio_title: "audio 2" }
	];

	const matchUrl = {
		url: "/podcast/0"
	};

	const component = renderer.create(
		<MemoryRouter>
			<PodcastAudiosList list={audios} match={matchUrl} />
		</MemoryRouter>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
