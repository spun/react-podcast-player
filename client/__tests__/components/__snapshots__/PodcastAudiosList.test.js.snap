// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcastAudiosList creation 1`] = `
<div
  className="card"
>
  <div
    className="header"
  >
    <h4
      className="title"
    >
      Podcast audios
    </h4>
  </div>
  <div
    className="content"
  >
    <ul
      className="list-unstyled team-members"
    >
      <li>
        <div
          className="row"
        >
          <div
            className="col-xs-1"
          >
            <div
              className="text-center"
              title="Invalid Date"
            >
              <span
                className="audio-date small "
              >
                Invalid Date
              </span>
              <br />
              <span
                className="audio-date small "
              >
                Invalid Date
              </span>
            </div>
          </div>
          <div
            className="col-xs-9"
          >
            <a
              className={undefined}
              href="/podcast/0/audios/1"
              onClick={[Function]}
              style={undefined}
            >
              audio 1
            </a>
            <br />
            <span
              className="text-muted audio-status"
              data-toggle="dropdown"
            >
              <small>
                Played 
                <span
                  className="caret"
                />
              </small>
            </span>
            <ul
              className="dropdown-menu"
            >
              <li
                onClick={[Function]}
              >
                <a
                  href="#"
                >
                  Mark as unplayed
                </a>
              </li>
              <li>
                <a
                  download={true}
                  href={undefined}
                >
                  Download file
                </a>
              </li>
            </ul>
          </div>
          <div
            className="col-xs-2 text-right"
          >
            <button
              className="btn btn-sm btn-success btn-icon"
              onClick={[Function]}
            >
              <i
                className="ti-control-play"
              />
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          className="row"
        >
          <div
            className="col-xs-1"
          >
            <div
              className="text-center"
              title="Invalid Date"
            >
              <span
                className="audio-date small "
              >
                Invalid Date
              </span>
              <br />
              <span
                className="audio-date small "
              >
                Invalid Date
              </span>
            </div>
          </div>
          <div
            className="col-xs-9"
          >
            <a
              className={undefined}
              href="/podcast/0/audios/2"
              onClick={[Function]}
              style={undefined}
            >
              audio 2
            </a>
            <br />
            <span
              className="text-muted audio-status"
              data-toggle="dropdown"
            >
              <small>
                Played 
                <span
                  className="caret"
                />
              </small>
            </span>
            <ul
              className="dropdown-menu"
            >
              <li
                onClick={[Function]}
              >
                <a
                  href="#"
                >
                  Mark as unplayed
                </a>
              </li>
              <li>
                <a
                  download={true}
                  href={undefined}
                >
                  Download file
                </a>
              </li>
            </ul>
          </div>
          <div
            className="col-xs-2 text-right"
          >
            <button
              className="btn btn-sm btn-success btn-icon"
              onClick={[Function]}
            >
              <i
                className="ti-control-play"
              />
            </button>
          </div>
        </div>
      </li>
    </ul>
    <div
      className=" footer text-center"
    >
      <button
        className="btn btn-info btn-fill btn-wd"
        onClick={[Function]}
        type="submit"
      >
        Load more audios
      </button>
    </div>
  </div>
</div>
`;
